crates/tessera-storage/src/shield/aes.rs

```diff
@@ -199,7 +211,9 @@ impl<S: Storage> AESShieldStorage<S> {
 }
 
 impl<S: Storage<Key = str, Value = [u8]>> Storage for AESShieldStorage<S> {
-    type StorageError = AESShieldStorageError;
+    // TODO: GATs would be nice here, but we can't use them
+    // because of the known limitation (https://github.com/rust-lang/rust/issues/100013)
+    type StorageError = AESShieldStorageError<'static>;
```

Comment:
@boris-cremit Do you have any good ideas for solving this issue?

Reply:
Sorry, I can't seem to come up with a good solution.